org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -703,4 +709,44 @@ public void projectsBuildFinished(IProgressMonitor monitor) {
 	public boolean isBuildFinished() {
 		return buildFinished;
 	}
+
+	@Override
+	public void checkIndexes() {
+		Job job = new Job(("Check Indexes")) {
+
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				IJavaProject[] javaProjects = ProjectUtils.getJavaProjects();
+				if (javaProjects.length <= 0) {
+					return Status.OK_STATUS;
+				}
+				SearchEngine engine = new SearchEngine();
+				IJavaSearchScope scope = SearchEngine.createWorkspaceScope();
+				boolean found[] = { false };
+				try {
+					engine.searchAllTypeNames("java.lang".toCharArray(), SearchPattern.R_EXACT_MATCH, "String".toCharArray(), //$NON-NLS-1$
+							SearchPattern.R_PATTERN_MATCH | SearchPattern.R_CASE_SENSITIVE, IJavaSearchConstants.CLASS, scope, new TypeNameRequestor() {
+								@Override
+								public void acceptType(int modifiers, char[] packageName, char[] simpleTypeName, char[][] enclosingTypeNames, String path) {
+									found[0] = true;
+								}
+							}, IJavaSearchConstants.WAIT_UNTIL_READY_TO_SEARCH, monitor);
+				} catch (JavaModelException e) {
+					// search failed
+				} catch (OperationCanceledException e) {
+					return Status.CANCEL_STATUS;
+				}
+				if (!found[0]) {
```

Comment:
I was thinking this also triggers the index rebuild when no VM has been configured, or maybe something went wrong there. However, I'm not sure we should rebuild the index in that case. Could we check that some VM is set as the default ? I was looking at https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/e90ec8f61d3b7918e4da8fc38a2d17eb4ff8893a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JVMConfigurator.java#L182-L193

Reply:
I'm fine with leaving this as-is for now.

Reply:
The SearchEngine.searchAllTypeNames method will find `java.lang.String` in some project's JDK, not only the default VM. It uses a workspace scope.